torch/_inductor/scheduler.py

```diff
@@ -2060,6 +2066,8 @@ def can_fuse_vertical(self, node1, node2):
         # we still can match unmet dep
         # if there's indirect indexing, don't match it
         def fusable_read_and_write(read: Dep, write: Dep):
```

Comment:
Side note: tracking `AccumulateDep` differently from `MemoryDep` should allow us to remove this hacky fix that blocks `atomic_add` from participating in vertical fusion 

https://github.com/pytorch/pytorch/blob/0c8a165b43c5cc226a84be46b8049e5dfca18aca/torch/_inductor/scheduler.py#L2073-L2090